### PR TITLE
SpriteSheet2D can be saved into xml file now

### PR DESCRIPTION
**Pull request by gleblebedev**
SpriteSheet2D can be saved into xml file now.
VectorBuffer GetName() returns AbstractFile::GetName() if set.

### CODE REVIEW

**Comment by rokups** [comment]
Legacy serialization will be deleted eventually. I wonder why you spent your precious time implementing that instead of new serialization..

File: Source/Urho3D/Urho2D/SpriteSheet2D.cpp [hunk 1]
```diff
@@ -51,6 +52,59 @@ void SpriteSheet2D::RegisterObject(Context* context)
...
+    if (rootElem.IsNull())
+    {
+        URHO3D_LOGERROR("Can not save SpriteSheet2D to null XML element");
+        return false;
+    }
+
+    if (texture_)
+    {
+        ea::string imagePath;
+        if (Urho3D::GetRelativePath(Urho3D::GetParentPath(GetName()), texture_->GetName(), imagePath))
```
Comment (rokups) [inline Source/Urho3D/Urho2D/SpriteSheet2D.cpp:75]: Why is this needed and what does it do? Looks odd. Seems like `texture_->GetName()` should always be enough.

Comment (gleblebedev) [reply Source/Urho3D/Urho2D/SpriteSheet2D.cpp:75]: It seems it is existing file format with a relative path to the texture: https://doc.starling-framework.org/v2.0/starling/textures/TextureAtlas.html

Comment (rokups) [reply Source/Urho3D/Urho2D/SpriteSheet2D.cpp:75]: Oh, this is implementing a third-party format. OK then, in that case my previous comment is not valid regarding new serialization. 👍

**Review by rokups — APPROVED** [review]
